synapse/storage/devices.py

```diff
@@ -825,6 +844,9 @@ def _add_device_change_txn(self, txn, user_id, device_ids, hosts, stream_id):
                     "device_id": device_id,
                     "sent": False,
                     "ts": now,
+                    "context": json.dumps(context)
+                    if whitelisted_homeserver(destination)
```

Comment:
could you factor this condition out to line 835 or so

Reply:
Sure

Reply:
Oh wait, no, I can't. It depends on the destination of the hosts